terraform-aws-github-runner/modules/runners/lambdas/runners/src/lambda.ts

```diff
@@ -1,24 +1,64 @@
-import { Context, SQSEvent, ScheduledEvent } from 'aws-lambda';
+import { SQS } from 'aws-sdk';
+import { Context, SQSEvent, SQSRecord, ScheduledEvent } from 'aws-lambda';
 
+import { Config } from './scale-runners/config';
 import { scaleDown as scaleDownR } from './scale-runners/scale-down';
-import { scaleUp as scaleUpR, RetryableScalingError } from './scale-runners/scale-up';
+import { scaleUp as scaleUpR, RetryableScalingError, ActionRequestMessage } from './scale-runners/scale-up';
+import { getDelayWithJitter } from './scale-runners/utils';
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export async function scaleUp(event: SQSEvent, context: Context, callback: any) {
   console.dir(event, { depth: 5 });
   try {
-    for (const e of event.Records) {
-      await scaleUpR(e.eventSource, JSON.parse(e.body));
+    const evtFailed: Array<SQSRecord> = [];
+
+    for (const evt of event.Records) {
+      try {
+        await scaleUpR(evt.eventSource, JSON.parse(evt.body));
+      } catch (e) {
+        if (e instanceof RetryableScalingError) {
+          console.error(`Retryable error thrown: "${e.message}"`);
+          evtFailed.push(evt);
+        } else {
+          throw e;
+        }
+      }
+    }
+
+    if (evtFailed.length > 0) {
+      console.error(`Detected ${evtFailed.length} errors when processing messages, will retry relevant messages.`);
+
+      const sqs: SQS = new SQS();
+
+      for (const evt of evtFailed) {
+        const body: ActionRequestMessage = JSON.parse(evt.body);
+        const retryCount = body?.retryCount ?? 0;
+        const delaySeconds = Math.max(body?.delaySeconds ?? Config.Instance.retryScaleUpRecordDelayS / 2, 10);
```

Comment:
Nit. 
The delay calculation formula looks a bit confusing. I'd rewrite it in a more explicit manner, like:

```js
          if (!body.delaySeconds) {
            body.delaySeconds = Config.Instance.retryScaleUpRecordDelayS;
          } else {
            body.delaySeconds *= 2;
          }
          body.delaySeconds = Math.max(body.delaySeconds, 10);
```

Reply:
I agree, much more readable this way